PEP-654: ExceptionGroup --> exception group in prose (#1855)
python/peps: pep-0654.rst

See the `Rejected Ideas`_ section for more on this.

The purpose of this PEP, then, is to add the ``except*`` syntax for handling
exception groups in the interpreter, which in turn requires that

> ``ExceptionGroup`` is added as a builtin type. The semantics of ``except*``
> are not backwards compatible with the current exception
> handling semantics, so we are not proposing to modify the behavior of the

Reviewer: While we're here, I would avoid the phrase "not backwards compatible" (since it's a red flag):
```suggestion
``ExceptionGroup`` is added as a builtin type. The desired semantics of
``except*`` are sufficiently different from the current exception
handling semantics that we are not proposing to modify the behavior of the
```

string and a sequence of the nested exceptions, which are exposed in the
fields ``message`` and ``errors``. For example:
``ExceptionGroup('issues', [ValueError('bad value'), TypeError('bad type')])``.
The difference between them is that ``ExceptionGroup`` can only wrap
``Exception`` subclasses while ``BaseExceptionGroup`` can wrap any
``BaseException`` subclass. A factory method that inspects the nested
execptions and selects between ``ExceptionGroup`` and ``BaseExceptionGroup``
makes the choice automatic. In the rest of the document, when we refer to
an exception group, we mean either an ``ExceptionGroup`` or a
``BaseExceptionGroup``. When it is necessary to make the distunction, we
use the class name. For brevity, we will use ``ExceptionGroup`` in code
examples that are relevant to both.

Since an exception group can be nested, it represents a tree of exceptions,
where the leaves are plain exceptions and each internal node represent a time
at which the program grouped some unrelated exceptions into a new group and
raised them together. The ``ExceptionGroup`` and ``BaseExceptionGroup``
classes are final, i.e., they cannot be further subclassed.

> 

Reviewer: ```suggestion
```
